### PR TITLE
[tune] Allow more versatile experiment analysis loading

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Collecting some initial feedback for better experiment analysis loading.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/trial.py
Comment (Yard1) [inline python/ray/tune/trial.py:238]: Could use a comment explaining its purpose

**Review by Yard1 — APPROVED** [review]
Looks great, thanks!

**Comment by krfricke** [comment]
Thanks, added a comment, tests passed before, so will merge after lint passed

**Pull request merged**
